tests/test_cli_compile.py

```diff
@@ -858,6 +858,27 @@ def test_upgrade_packages_option_no_existing_file(pip_conf, runner):
     assert out.exit_code == 0
     assert "small-fake-a==0.2" in out.stderr.splitlines()
     assert "small-fake-b==0.3" in out.stderr.splitlines()
+    assert (
+        "WARNING: the output file requirements.txt exists but is empty"
+        not in out.stderr
+    )
+
+
+def test_upgrade_packages_option_empty_existing_file_warning(pip_conf, runner):
```

Comment:
```suggestion
def test_upgrade_packages_empty_target_file_warning(pip_conf, runner):
```
Maybe this sounds better.